docs/theory/diffusion/diffusion.rst

```diff
@@ -0,0 +1,667 @@
+
+    fig = plt.figure(figsize=(10, 5))
+    ax = fig.add_subplot(121, projection='3d')
+    view_limit = 0.8
+    for lag_time, hist in zip(frame_lags, msd_histogram):
+        ax.bar(
+            histogram_centers[histogram_centers < view_limit],
+            hist[histogram_centers < view_limit],
+            zs=lag_time,
+            zdir='y',
+            alpha=0.8,
+            width=histogram_centers[1]-histogram_centers[0]
+        )
+
+    plt.xlim([0, 0.8])
+    ax.view_init(20, -60, 0)
+    ax.set_zticks([])
+    ax.set_xlabel(r'MSD [$\mu m^2$]')
+    ax.set_ylabel('Lag time [s]')
+    plt.tight_layout();
+
+.. image:: figures/msd_distributions.png
+
+If a group of tracks experience the same environment and diffusion coefficient, then it makes sense to compute an averaged MSD estimate using :meth:`~lumicks.pylake.kymotracker.kymotrack.KymoTrackGroup.ensemble_msd`::
+
+    tracks = lk.simulation.simulate_diffusive_tracks(diffusion_constant=1.0, steps=40, dt=0.01, observation_noise=0.1, num_tracks=25)
+    ensemble_msd = tracks.ensemble_msd()
+
+This returns a weighted average of the MSDs coming from all the tracks (where the weight is determined by the number of points that contribute to each estimate).
+If the tracks are of equal length, this weighting will not have an effect (since all the weights will be the same).
+You can plot the ensemble msd as follows::
+
+    ensemble_msd.plot()
+
+.. image:: figures/ensemble_msd.png
+
+MSD estimates should be handled with care.
+It is important to remember that MSDs for large lags will show a high variance and should not be assumed to be Gaussian distributed.
+In addition to that, MSDs for different lags are highly correlated.
+One should not fit these values as though they were independent data points unless care is taken that it is valid to do so (more on this below).
+
+Ordinary Least Squares
+^^^^^^^^^^^^^^^^^^^^^^
+
+Real measurements are affected by noise, which leads to localization error.
+In addition to this, depending on the method used to detect the position of the particle, there may also be motion blur.
+These two sources of error manifest themselves as an offset in the MSD curve given by:
+
+.. math::
+
+    offset = 2 \sigma^2 - 4 R D \Delta t
+
+Here :math:`\sigma` is the static localization accuracy, :math:`R` is a motion blur constant and :math:`\Delta t` represents the time step.
+With pure diffusive motion (a complete absence of drift) in an isotropic medium, 1-dimensional MSDs can be fitted by the following relation:
+
+.. math::
+
+    \rho[n] = 2 D n \Delta t + offset
+
+Here :math:`D` is the diffusion constant in :math:`um^2/s`, :math:`\Delta t` is the time step, :math:`n` is the step index and the offset is determined by the localization accuracy.
+
+While it may be tempting to use a large number of lags in the diffusion estimation procedure, this actually produces poor estimates of the diffusion constant :cite:`qian1991single,michalet2010mean,michalet2012optimal`.
```

Comment:
```suggestion
While it may be tempting to use a large number of lags in the diffusion estimation procedure, this actually produces poor estimates of the diffusion constant :cite:`qian1991single,michalet2010mean,michalet2012optimal`, because, as mentioned above, the error in the MSD value increases with lag time.

```

Reply:
👍